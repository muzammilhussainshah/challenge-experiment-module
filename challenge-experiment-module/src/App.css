.App {
  margin: 20px;
}

.container {
  width: 600px;
  border-radius: 10px;
  align-self: center;
  margin: 10px auto;
  box-shadow: 0px 0px 10px 4px rgba(0, 0, 0, 0.6);
}

.header {
  height: 50px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 20px;
  background-color: rgba(25, 25, 25, 1);
  /* borderRadius: `10px`, */
  border-radius: 10px;
}

.header h3 {
  color: #000;
}

.content {
  background-color: rgba(25, 25, 25, 1);
  width: 100%;
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  padding: 16px 0;
}

body {
  background-color: rgba(117, 117, 117, 1);
}

.title {
  color: rgba(117, 117, 117, 1);
}

.flipped-icon {
  font-size: 24px;
  color: rgba(117, 117, 117, 1) !important;
}

.input_container {
  background-color: black;
  min-height: 44px;
  display: flex;
  align-items: center;
  width: 95%;
  border-radius: 10px;
  margin: 0px auto 5px auto;
}

.input_title {
  font-size: 16px;
  margin: 0 16px;

  color: rgba(117, 117, 117, 1);
  background-color: transparent;
}

.custom-input::placeholder {
  color: #999;
  font-size: 16px;
  margin: 0 16px;
  color: rgba(117, 117, 117, 1);
}

.custom-input:focus {
  outline: none;
  border-bottom: none;
}

.custom-input {
  color: white;
  background: transparent;
  border: none;
  margin: 0 16px;
  padding: 0;
  width: 75%;
}

.under_line {
  text-decoration: underline;
}

.prompt_container {
  background-color: black;
  display: flex;
  padding: 16px 0;
  align-items: center;
  width: 95%;
  border-radius: 10px;
  margin: 16px auto;
}

.footer_buttons_container {
  display: flex;
  margin: 15px 0 0 0;
  justify-content: end;
  text-transform: uppercase;
  font-weight: 700;
}

.footer_buttons:hover {
  color: white;
  cursor: pointer;
}

.generate_button:hover {
  cursor: pointer;
}


.generate_module_buttons {
  height: 30px;
  display: flex;
  font-size: 13px;
  justify-content: center;
  padding: 0px 10px;
  border-radius: 10px;
  text-transform: uppercase;
  align-items: center;

  border: 1px solid rgba(117, 117, 117, 1);
  color: rgba(117, 117, 117, 1);
}

.generate_module_buttons:hover {
  cursor: pointer;
}

.generate_module_buttons_container {
  display: flex;
  gap: 10px;
  border-bottom: 1px solid #191919;
  padding: 10px;
  flex-wrap: wrap;

}

.activeButton {

  cursor: pointer;
  border: 1px solid green;
  color: green;
}

summary::marker {
  content: '';
}

details {
  overflow: hidden;
  border-radius: 10px;
}

.contentContainer {
  /* margin: '5px auto',  */
  margin: 5px auto;
}
.input_title_adjustment{
  margin: 16px;
}