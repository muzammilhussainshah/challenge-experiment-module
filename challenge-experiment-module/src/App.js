import React, { useState } from 'react'

import './App.css'
import { ExperimentWidget } from './components/experiment-widget.tsx'

function App () {
  const [experiments, setExperiments] = useState([])

  const handleExperiment = (value) => {
    const key = Object.keys(value)[0]
    const experimentsClone = experiments.slice()
    const existingIndex = experimentsClone.findIndex((item) => Object.keys(item)[0] === key)
    if (existingIndex === -1) setExperiments([...experimentsClone, value])
    else {
      experimentsClone[existingIndex] = value
      setExperiments(experimentsClone)
    }
  }
  const handleCreateExperiment = (value) => setExperiments([...experiments, value])
  return (
    <div className='App'>

      {/* CREATE EXPERIMENT MODULE */}
      <ExperimentWidget createModule callback={handleCreateExperiment} />

      {/* EXPERIMENT MODULE LIST */}
      {experiments?.length > 0 && experiments?.map((item, index) => (
        <ExperimentWidget
          key={index.toString()}
          experimentalModuleCartDataProps={{
            id: Object.keys(item)[0],
            experiments: item[Object.keys(item)[0]]
          }}
          callback={handleExperiment}
        />
      ))}
    </div>
  )
}

export default App
