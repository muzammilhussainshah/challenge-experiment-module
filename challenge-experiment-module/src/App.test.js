import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import App from './App'

describe('App Component', () => {
  // Test to ensure that the App component renders without crashing
  test('renders without crashing', () => {
    render(<App />)
  })

  // Test to ensure that the "Create Experiment Module" is rendered
  test('renders Create Experiment Module', () => {
    const { getByText } = render(<App />)
    const createModule = getByText('Create Experiment Module')
    expect(createModule).toBeInTheDocument()
  })

  // Test to ensure that the "Experiment Module List" is rendered
  test('renders Experiment Module List', () => {
    const { getByText } = render(<App />)
    const experimentList = getByText('Experiment Module')
    expect(experimentList).toBeInTheDocument()
  })
})
