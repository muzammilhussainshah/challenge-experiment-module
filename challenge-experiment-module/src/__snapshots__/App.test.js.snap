// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="container"
    >
      <div
        class="header rounded-bottom"
      >
        <h3
          class="title"
          style="color: rgb(117, 117, 117);"
        >
          Create Experiment Module
        </h3>
      </div>
    </div>
  </div>
</div>
`;
